### PR TITLE
Added note about `quarkus.native.container-build=true` 

### PR DESCRIPTION
**Pull request by hbelmiro**
This PR improves the documentation related to building native images.
Context: https://github.com/quarkusio/quarkus/issues/27603#issuecomment-1233232027

### CODE REVIEW

**Review by galderz — APPROVED** [review]

File: docs/src/main/asciidoc/building-native-image.adoc [hunk 1]
```diff
@@ -424,6 +424,12 @@ If one of those extensions is present, then creating a container image for the n
 ----
 
 * `quarkus.native.container-build=true` allows for creating a Linux executable without GraalVM being installed (and is only necessary if you don't have GraalVM installed locally or your local operating system is not Linux)
+
+[NOTE]
+====
+If you're running a remote Docker daemon, you need to replace `quarkus.native.container-build=true` with `quarkus.native.remote-container-build=true`.
```
Comment (zakkak) [inline docs/src/main/asciidoc/building-native-image.adoc:430]: This is already mentioned in https://github.com/quarkusio/quarkus/blob/b9ba708f477d2aec1873d4f4e041e061793e49a7/docs/src/main/asciidoc/building-native-image.adoc#creating-a-linux-executable-without-graalvm-installed (2nd tip).

I suggest adding a reference to that tip as it contains more details about why this is necessary.

**Review by zakkak — APPROVED** [review]
It looks good to me, but IMO it could be improved by adding a reference to the more detailed "tip" as mentioned in my comment.

Thanks

File: docs/src/main/asciidoc/building-native-image.adoc (hunk 1 above)
Comment (hbelmiro) [reply docs/src/main/asciidoc/building-native-image.adoc:431]: Thanks for your review, @zakkak.
I've added the suggested change.

**Comment by geoand** [comment]
Mind squashing the commits please?

**Review by geoand — APPROVED** [review]

**Pull request merged**